test/unit_tests/test_braket_device.py

```diff
@@ -2130,3 +2131,105 @@ def __init__(self):
         assert settings["wires"] == [0, 1, 2, 3, 4, 5, 6, 7]
         assert np.allclose(settings["qubit_freq"], 4.6)
         assert np.allclose(settings["anharmonicity"], 0.1)
+
+
+def get_device():
+    dev = _aws_device(wires=2, device_arn="arn:aws:braket:eu-west-2::device/qpu/oqc/Lucy")
+
+    class DummyProperties:
+        def __init__(self):
+            self.pulse = PulseDeviceActionProperties.parse_raw(OQC_PULSE_PROPERTIES)
+            self.paradigm = GateModelQpuParadigmProperties.parse_raw(OQC_PARADIGM_PROPERTIES)
+
+    dev._device._properties = DummyProperties()
+
+    return dev
+
+
+class TestPulseValidation:
+    def test_validate_hamiltonian_settings_raises_a_warning(self):
+        """Check that a warning is raised if the settings from the interaction term
+        on the ParametrizedEvolution don't match the device constants"""
+
+        dev = get_device()
+
+        # some 3 qubit device
+        H = qml.pulse.transmon_interaction(
+            qubit_freq=[4.3, 4.6, 4.8],
+            connections=[(1, 2), (1, 3)],
+            coupling=[0.02, 0.03],
+            wires=[0, 1, 2],
+        )
+        # 4.3 GHz drive on wire 0 with phase=0 and amplitude=0.2
+        H += qml.pulse.transmon_drive(0.2, 0, 4.3, wires=[0])
+
+        op = ParametrizedEvolution(H, [], t=10)
+
+        with pytest.warns(UserWarning) as record:
+            dev._validate_hamiltonian_settings(op)
+
+        # check that only one warning was raised
+        assert len(record) == 1
+        # check that the message matches
+        assert record[0].message.args[0][-25:] == "do not match the hardware"
```

Comment:
That looks a bit unstable, can we not do
```python
with pytest.warns(UserWarning, match="do not match the hardware"):
    dev._validate_hamiltonian_settings(op)
```
instead?

Reply:
Sure, I can switch it to that :)